beacon-chain/blockchain/receive_attestation.go

```diff
@@ -18,10 +18,16 @@ import (
 	"go.opencensus.io/trace"
 )
 
+// AttestationStateFetcher is able to retrieve a beacon state for an attestation
+// target checkpoint, useful for verifying attestation signatures.
+type AttestationStateFetcher interface {
+	AttestationPreState(ctx context.Context, attTarget *ethpb.Checkpoint) (*state.BeaconState, error)
```

Comment:
I just realized `AttestationPreState` is a poor choice for name from before. Can we rename it to `AttestationTargetState`?